OSDOCS#4563: Add common terms for registry book (#52812)
openshift/openshift-docs: modules/registry-common-terms.adoc

// Module included in the following assemblies:
//
// * registry/index.adoc

:_content-type: REFERENCE
[id="openshift-registry-common-terms_{context}"]
= Common registry terms

> This glossary defines common terms that are used in the registry content. 

Reviewer: `the common terms`

Author: acked

> container::

Reviewer 2: Better to use uppercase letter for these iterms

Author: As per IBM style guide suggestion[1] we should keep them lower case letters 

[1] `Do not capitalize a term in the glossary unless it is a proper noun.`

> Lightweight and executable images that consist software and all its dependencies. Because containers virtualize the operating system, you can run containers anywhere, from a data center to a public or private cloud to your local host.

Reviewer: run `the` containers

Author: acked

> Lightweight and executable images that consist software and all its dependencies. Because containers virtualize the operating system, you can run containers anywhere, from a data center to a public or private cloud to your local host.

Reviewer: `you can run containers anywhere, from a data center to a public or private cloud to your local host.` 
Can you formalize this sentence?
Suggestion 1: `you can run containers anywhere.`
Suggestion 2: `you can run containers in data center, a public or private cloud, or your local host.` 

You can also come up with something better if you feel so.

Author: acked

Image Registry Operator::
The Image Registry Operator runs in the `openshift-image-registry` namespace, and manages the registry instance in that location.

image repository::
An image repository is a collection of related container images and tags identifying images.

mirror registry::

> A registry that holds the mirror of {product-title} images.

Reviewer: You can start with `The mirror registry is a registry....` to stay consistent with the other definitions or you can modify other terms to be consistent with this one.

Author: acked

namespace::

> A namespace isolates specific system resources that are visible to all processes. Inside a namespace, only processes that are members of that namespace can see those resources.

Reviewer 3: where do we use this term in the registry docs?

Author: A term namespace is mentioned under https://docs.openshift.com/container-platform/4.12/registry/configuring_registry_storage/configuring-registry-storage-vsphere.html

Reviewer 3: This definition seems to be about kernel namespaces - https://en.wikipedia.org/wiki/Linux_namespaces
but we usually use `namespace` as a Kubernetes namespace - https://kubernetes.io/docs/concepts/overview/working-with-objects/namespaces/ - on k8s level there are no processes

Author: acked

{product-title} registry::

> {product-title} registry is a registry provided by {product-title} to manage images.

Reviewer 4: ```suggestion
{product-title} registry is the registry provided by {product-title} to manage images.
```

Author: acked

pod::
The pod is the smallest logical unit in Kubernetes. A pod is comprised of one or more containers to run in a worker node.

private registry::

> A registry is a server that implements the container image registry API. A private registry is a registry that requires authentication to allow users access to its contents.

Reviewer: ```suggestion
A registry is a server that implements the container image registry API. A private registry is a registry that requires authentication to allow users access its contents.
```

Author: acked

> private registry::
> {product-title} can use any server implementing the container image registry API as a source of the image which allows the developers to push and pull their private container images.
> 
> public registry::
> {product-title} can use any server implementing the container image registry API as a source of the image which allows the developers to push and pull their public container images.

Reviewer 4: This section confused me a bit, so I gave a stab at clarifying it - feel free to use it as is or as inspiration.

```suggestion
private registry::
A registry is a server that implements the container image registry API. A private registry is a registry that requires authentication to allow users access to its contents.

public registry::
A registry is a server that implements the container image registry API. A public registry is a registry that serves its contently publicly.
```

Author: acked

> {product-title} can use any server implementing the container image registry API as a source of the the image which allows the developers to push and pull their public container images.

Reviewer 2: Duplicate `the`

Author: acked

> {op-system-base} Quay Container Registry::

Reviewer 2: s/{op-system-base}/Redhat

Author: We must use {op-system-base} for the term Red Hat in the docs

Reviewer 2: But in https://52812--docspreview.netlify.app/openshift-enterprise/latest/registry/index.html#openshift-registry-common-terms_registry-overview,  it's `RHEL Quay Container Registry`, it's not correct

Author: oh I see. I think you are right. I changed it to `Red Hat`

> A Quay container registry that serves most of the container images and Operators to {product-title} clusters.

Author: yes

> Red Hat Quay Container Registry::
> A Quay container registry that serves most of the container images and Operators to {product-title} clusters.

Reviewer 4: ```suggestion
Quay.io::
A public Red Hat Quay Container Registry instance provided and maintained by Red Hat, that serves most of the container images and Operators to {product-title} clusters.
```

Author: acked

> A Quay container registry that serves most of the container images and Operators to {product-title} clusters.

Reviewer 4: Gotcha - suggested an edit in an attempt to clarify the relationship between Quay.io and Red Hat Quay Container Registry

> A Quay container registry that serves most of the container images and Operators to {product-title} clusters.

Author: acked

> A Quay.io container registry that serves most of the container images and Operators to {product-title} clusters.

Reviewer 2: s/Quay.io/Quay

Author: acked

registry authentication::
To push and pull images to and from private image repositories, The registry needs to authenticate its users with credentials.

route::
Exposes a service to allow for network access to pods from users and applications outside the {product-title} instance.

scale down::

> Decreasing the number of replicas.

Reviewer: `To decrease the number of replicas`
Any particular reason why continuous tense is used?

Author: acked

scale up::

> Increasing the number of replicas.

Reviewer: Same as scale down

Author: acked

> {product-title} can communicate with registries to access private image repositories using credentials supplied by the user. This allows {product-title} to push and pull images to and from private repositories

Reviewer 3: it doesn't look like a definition of "registry authentication". Maybe something like:

> To push and pull images to and from private image repositories, the registry needs to authenticate its users with credentials.

> Route::

Reviewer 2: Need lowercase

Author: acked

> Decreasing the resource capacity.

Author: acked

> Decreasing the resource capacity.

Reviewer 2: Scale down, scale up and Service
Do we need them here?

Author: Yes. I think we should keep them here. Because these terms come across in the registry book and customers will understand what these terms mean

private registry::
{product-title} can use any server implementing the container image registry API as a source of the image which allows the developers to push and pull their private container images.

public registry::
{product-title} can use any server implementing the container image registry API as a source of the the image which allows the developers to push and pull their public container images.

{op-system-base} Quay Container Registry::
A Quay.io container registry that serves most of the container images and Operators to {product-title} clusters.

registry authentication::
{product-title} can communicate with registries to access private image repositories using credentials supplied by the user. This allows {product-title} to push and pull images to and from private repositories

Route::
Exposes a service to allow for network access to pods from users and applications outside the {product-title} instance.

scale down::
Decreasing the resource capacity.

scale up::
Increasing the resource capacity.

service::
A service exposes a running application on a set of pods.

> user-provisioned infrastructure::

Reviewer 2: We don't need this

Author: acked

Author: acked

// Module included in the following assemblies:
//
// * registry/index.adoc

:_content-type: REFERENCE
[id="openshift-registry-common-terms_{context}"]
= Common terms

This glossary defines common terms that are used in the architecture content. These terms help you understand {product-title} architecture effectively.

> authentication::

Reviewer 2: Should we use registry authentication here?
```
OpenShift Container Platform can communicate with registries to access private image repositories using credentials supplied by the user. This allows OpenShift Container Platform to push and pull images to and from private repositories
```

Author: Addressed

To control access to an {product-title} cluster, a cluster administrator can configure user authentication and ensure only approved users access the cluster. To interact with an {product-title} cluster, you must authenticate to the {product-title} API. You can authenticate by providing an OAuth access token or an X.509 client certificate in your requests to the {product-title} API.

container::
Lightweight and executable images that consist software and all its dependencies. Because containers virtualize the operating system, you can run containers anywhere, from a data center to a public or private cloud to your local host.

> custom resource definition (CR)::

Reviewer 2: IMO, we don't need this

Author: acked

The Image Registry Operator runs in the `openshift-image-registry` namespace, and manages the registry instance in that location.

metrics::

> Metrics tracks the utilization and availability of resources such as CPU, memory, and storage.

Reviewer 2: IMO, we don't need metrics here

Author: acked

> 

Reviewer 2: ```
Image repository
An image repository is a collection of related container images and tags identifying them
```

Author: Addressed
